menu.h

```diff
@@ -0,0 +1,103 @@
+#include <iostream>
+#include "utils.h"
+
+using namespace std;
+namespace menu
+{
+    void menu()
+    {
+        cout << "Welcome to the game!" << endl;
+        cout << "Please select an option:" << endl;
+        cout << "1. Start Game" << endl;
+        cout << "2. Settings" << endl;
+        cout << "3. Quit" << endl;
+        int textsize=16;
+        int weightsize=400;
+        int choice;
+        cin >> choice;
+
+        switch (choice)
+        {
+        case 1:
+        {
+            // code for starting the game
+            cout << "Starting the game..." << endl;
+            break;
+        }
+        case 2:
+        {
+            // code for settings
+            cout << "Opening settings..." << endl;
+            cout << "chouse a setting you want to change" << endl;
+            cout << "Please select an option:" << endl;
+            cout << "1. Text Size" << endl;
+            cout << "2. Font Weight" << endl;
+            cout << "3. Face Name" << endl;
+            int Setting_shoice;
+            cin >> Setting_shoice;
+            switch (Setting_shoice)
+            {
+            case 1:
+            {
+                cout << "Pick Number" << endl;
+                cin >> textsize;
+                utils::customize_text(textsize);
+                menu();
+                break;
+            }
+            case 2:
+            {
+                cout << "Pick Weight" << endl;
+                cout << "1. light" << endl;
+                cout << "2. SemiLight" << endl;
+                cout << "3. Normal" << endl;
+                cout << "4. SemiBold" << endl;
+                cout << "5. Bold" << endl;
+                cin >> weightsize;
+                if (weightsize == 1)
+                {
+                    utils::customize_text(textsize, utils::Light);
+                }
+                else if (weightsize == 2)
+                {
+                    utils::customize_text(textsize, utils::SemiLight);
+                }
+                else if (weightsize == 3)
+                {
+                    utils::customize_text(textsize, utils::Normal);
+                }
+                else if (weightsize == 4)
+                {
+                    utils::customize_text(textsize, utils::SemiBold);
+                }
+                else if (weightsize == 5)
+                {
+                    utils::customize_text(textsize, utils::Bold);
+                }
+                menu();
+                break;
+            }
+            case 3:
+            {
+                cout << "no idea what to do..." << endl;
+                menu();
+                break;
+            }
+            }
+        }
+        case 3:
+        {
+            // code for quitting
+            cout << "Quitting the game..." << endl;
+            exit(0);
```

Comment:
This function does not do proper cleanup (in particular, it does not call the destructor of any automatic-storage object), and it shouldn't be used. The proper way to terminate the program is to `return 0` from `main()`. Read more here: https://stackoverflow.com/questions/30250934/how-to-end-c-code

Reply:
finaly did this one too